Merge the two commented-out alternatives to the 'Message Format' section (#120)
ircdocs/modern-irc: index.md

### Tags

> The **tags** part is optional, and MUST NOT be sent unless explicitly enabled by a [capability](#capability-negotiation). Messages may omit the part entirely. This message part starts with a leading `('@', 0x40)` character, which MUST be the first character of the message itself. The leading `('@', 0x40)` is stripped from the value before it's processed further.

Reviewer: We don't really need to re-state how optional this part is because of how clearly we're stating it now with that MUST NOT.

```suggestion
The **tags** part is optional, and MUST NOT be sent unless explicitly enabled by [a capability](#capability-negotiation). This message part starts with a leading `('@', 0x40)` character, which MUST be the first character of the message itself. The leading `('@', 0x40)` is stripped from the value before it gets processed further.
```

### Source

>       source          ::=  servername / ( nickname [ [ "!" user ] "@" host ] )

Reviewer 2: ```suggestion
      source          ::=  servername / ( nickname [ "!" user ] [ "@" host ] )
```

RFC 2812 broke compatibility with RFC 1459 here for no apparent reason.

Author: Right.

I think I'll send a PR in the future to say servers SHOULD send user and host, because they rarely do that, it's very annoying for clients, and the only use case I have ever seen is `draft/relaymsg`.
